pkg/server/api/entry/v1/service.go

```diff
@@ -351,14 +359,18 @@ func (s *Service) isEntryUnique(ctx context.Context, e *common.RegistrationEntry
 			Selectors: e.Selectors,
 		},
 	})
+
 	if err != nil {
-		return api.MakeStatus(log, codes.Internal, "failed to list entries", err)
+		return nil, false, err
 	}
-	if len(resp.Entries) != 0 {
-		return api.MakeStatus(log, codes.AlreadyExists, "entry already exists", nil)
+
+	existing := new(common.RegistrationEntry)
```

Comment:
nitpick: I wonder if we shouldn't return `nil` for the entry when it does not exist. Having a non-nil entry seems a little more dangerous since it may not be as apparent if it is accidentally used.